internal/provider/resource_gitlab_project_environment_test.go

```diff
@@ -0,0 +1,221 @@
+package provider
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabProjectEnvironment_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	var env gitlab.Environment = gitlab.Environment{
+		Name: fmt.Sprintf("ProjectEnvironment-%d", rInt),
+	}
+	var env2 gitlab.Environment = gitlab.Environment{
+		Name:        fmt.Sprintf("ProjectEnvironment-%d", rInt),
+		ExternalURL: "https://example.com",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectEnvironmentDestroy,
+		Steps: []resource.TestStep{
```

Comment:
Please test imports, see: https://github.com/gitlabhq/terraform-provider-gitlab/blob/32e9fe228f6d0a3b8feafe407f2286b36a55f927/internal/provider/resource_gitlab_project_issue_test.go#L46

There is actually value in testing that after every step, to check if the state setting is consistent with the API responses ...